### ISSUE TITLE
Create abstraction of `survey.Validator`

### ISSUE THREAD

**Issue by craicoverflow** [redhat-developer/app-services-cli#722]
We are using a localization interface everywhere now which needs to be injected into any component which needs to be localized.
One such place that needs this is our validator functions which bind to signature of https://github.com/AlecAivazis/survey/blob/v2.2.12/survey.go#L93.

To work around this we are using higher-order functions which return an anonymous function to execute the validation.

```go
validate := ValidateName(localizer)(value)
```

Readibility would be improved if we create a validator abstraction with a named function instead which when implemented would look like this:

```go
validator := serviceaccountutil.Validator{Localizer: opts.localizer}
validator.ValidateName()
validator.ValidateUUID()
validator.ValidateX()
```

This will greatly improve readibility and will keep validation logic organised.

Let me know if you have any questions on this @rkpattnaik780

**Comment by rkpattnaik780** [comment]
This is a very neat approach! Thanks for bringing it up :) .

**Comment by craicoverflow** [comment]
Can this be closed?

**Comment by rkpattnaik780** [comment]
Closed by #721
